client/src/main/java/tds/config/ClientSystemFlag.java

```diff
@@ -1,82 +1,144 @@
+            this.dateChanged = dateChanged;
+            return this;
+        }
 
-    public ClientSystemFlag() { }
+        public Builder withDatePublished(Instant datePublished) {
+            this.datePublished = datePublished;
+            return this;
+        }
 
-    public ClientSystemFlag(String auditObject, String clientName) {
-        this.setAuditObject(auditObject);
-        this.setClientName(clientName);
+        public ClientSystemFlag build() {
+            return new ClientSystemFlag(this);
+        }
     }
 
-    public String getAuditObject() {
-        return auditObject;
+    public ClientSystemFlag() {}
+
+    private ClientSystemFlag(Builder builder) {
+        this.auditObject = builder.auditObject;
+        this.clientName = builder.clientName;
+        this.isPracticeTest = builder.isPracticeTest;
+        this.isOn = builder.isOn;
+        this.description = builder.description;
+        this.dateChanged = builder.dateChanged;
+        this.datePublished = builder.datePublished;
     }
 
-    public void setAuditObject(String auditObject) {
-        this.auditObject = auditObject;
+
+    /**
+     * @return The type of thing this {@link ClientSystemFlag} represents/is related to.
+     */
+    public String getAuditObject() {
+        return auditObject;
     }
 
+    /**
+     *  @return The name of the client that owns this {@link ClientTestProperty}.
+     */
     public String getClientName() {
         return clientName;
     }
 
-    public void setClientName(String clientName) {
-        this.clientName = clientName;
-    }
-
-    public Boolean getIsPracticeTest() {
+    /**
+     * @return Identify if the {@link ClientSystemFlag} is associated/relevant to practice assessments.
+     */
+    public boolean getIsPracticeTest() {
         return isPracticeTest;
     }
 
-    public void setIsPracticeTest(Boolean practiceTest) {
-        isPracticeTest = practiceTest;
-    }
-
-    public Boolean getIsOn() {
+    /**
+     * <p>
+     *     Cannot be null in {@code configs.client_systemflags} table.  This value is an  {#code INT(11)} in the
+     *     database, but is treated like a boolean value in the legacy code's logic.  An example of how this property is
+     *     treated (edited for clarity):
+     *     <pre>
+     *         {@code
+     *             select IsOn as flag from  ${ConfigDB}.client_systemflags F, externs E  where E.ClientName = ${clientname} and F.clientname = ${clientname} and E.IsPracticeTest = F.IsPracticeTest and AuditOBject = ${responses}
+     *             // ... execute query...
+     *             if (DbComparator.isEqual (flag, 0)) {
+     *                 return 0;
+     *             } else {
+     *                 return 1;
+     *             }
+     *         }
+     *     </pre>
+     * </p>
+     *
+     * @return {@code True} if the {@link ClientSystemFlag} is enabled; otherwise {@code False}.
+     */
+    public boolean getIsOn() {
```

Comment:
Java convention for boolean getters usually start with "isSomething".  I know this is odd because they've literally named the field in the DB "isOn".  I actually would prefer 'on' (or something more descriptive) be the name and this be isOn().  I think keeping names from the DB in our Java isn't necessary if we can come up with better names.